cmd/step/sync_steps.go

```diff
@@ -0,0 +1,44 @@
+package step
+
+import (
+  "fmt"
+
+  "github.com/Originate/gt/cmd/config"
+  "github.com/Originate/gt/cmd/git"
+)
+
+
+func GetSyncBranchSteps(branchName string) []Step {
+  isFeature := config.IsFeatureBranch(branchName)
+  hasRemote := config.HasRemote()
+
+  var steps []Step
+
+  if hasRemote || isFeature {
+    steps = append(steps, CheckoutBranchStep{BranchName: branchName})
+    if isFeature {
+      steps = append(steps, MergeTrackingBranchStep{}, MergeBranchStep{BranchName: config.GetParentBranch(branchName)})
+    } else {
+      if config.GetPullBranchStrategy() == "rebase" {
+        steps = append(steps, RebaseTrackingBranchStep{})
+      } else {
+        steps = append(steps, MergeTrackingBranchStep{})
```

Comment:
What do you think about creating a struct called `StepList` that allows for a slightly nicer API to fill it:

```go
steps.append(MergeTrackingBranchStep{})
```

Then we can add other methods like `prepend`, `replace`, etc